src/alire/alire-publish.adb

```diff
@@ -1018,6 +1093,12 @@ package body Alire.Publish is
            ("URL seems to point to a repository, but no commit was provided.");
       end if;
 
+      --  Verify that subdir is not used with an archive (it is only for VCSs)
+
+      if Subdir /= "" and then Commit = "" then
+         Raise_Checked_Error ("Cannot publish a nested crate from an archive");
```

Comment:
Why do we have this limitation?

Reply:
I wanted to tackle the presumed more wanted feature first. In principle it should be doable too, but I'm not sure it makes sense to publish as a source archive and include folders over the one you want to publish. Maybe to share resources at some point?